packages/mask/src/extension/background-script/EthereumServices/request.ts

```diff
@@ -1,261 +1,81 @@
-import type { RequestArguments, TransactionConfig } from 'web3-core'
-import type { JsonRpcPayload, JsonRpcResponse } from 'web3-core-helpers'
-import {
-    EthereumMethodType,
-    EthereumTransactionConfig,
-    formatGweiToWei,
-    formatWeiToGwei,
-    isEIP1559Supported,
-    ProviderType,
-    RequestOptions,
-    SendOverrides,
-    TransactionStateType,
-} from '@masknet/web3-shared-evm'
-import {
-    currentChainIdSettings,
-    currentMaskWalletAccountSettings,
-    currentMaskWalletChainIdSettings,
-    currentProviderSettings,
-} from '../../../plugins/Wallet/settings'
-import { WalletRPC } from '../../../plugins/Wallet/messages'
-import { INTERNAL_nativeSend, INTERNAL_send } from './send'
-import { defer } from '@dimensiondev/kit'
-import { hasNativeAPI, nativeAPI } from '../../../../shared/native-rpc'
-import { openPopupWindow } from '../../../../background/services/helper'
-import Services from '../../service'
-import { toHex } from 'web3-utils'
-import { isLessThan } from '@masknet/web3-shared-base'
-
-let id = 0
-
-const UNCONFIRMED_CALLBACK_MAP = new Map<number, (error: Error | null, response?: JsonRpcResponse) => void>()
-const RISK_METHOD_LIST = [
-    EthereumMethodType.ETH_SIGN,
-    EthereumMethodType.PERSONAL_SIGN,
-    EthereumMethodType.ETH_SIGN_TYPED_DATA,
-    EthereumMethodType.ETH_DECRYPT,
-    EthereumMethodType.ETH_GET_ENCRYPTION_PUBLIC_KEY,
-    EthereumMethodType.ETH_SEND_TRANSACTION,
-]
-
-function getSendMethod() {
-    if (hasNativeAPI && nativeAPI) return INTERNAL_nativeSend
-    return INTERNAL_send
-}
-
-function getPayloadId(payload: JsonRpcPayload) {
-    return typeof payload.id === 'string' ? Number.parseInt(payload.id, 10) : payload.id
-}
-
-function isRiskMethod(method: EthereumMethodType) {
-    return RISK_METHOD_LIST.includes(method)
-}
-
-function isSendMethod(method: EthereumMethodType) {
-    return method === EthereumMethodType.ETH_SEND_TRANSACTION
+import type { RequestArguments } from 'web3-core'
+import type { JsonRpcPayload } from 'web3-core-helpers'
+import { EthereumMethodType, ProviderType, RequestOptions, SendOverrides } from '@masknet/web3-shared-evm'
+import { createExternalProvider } from './provider'
+import { createContext, dispatch, use } from './composer'
+
+// #region middleware
+import { Squash } from './middleware/Squash'
+import { Nonce } from './middleware/Nonce'
+import { Interceptor } from './middleware/Interceptor'
+import { Popup } from './middleware/Popup'
+import { Translator } from './middleware/Translator'
+import { RecentTransaction } from './middleware/Transaction'
+import { TransactionNotifier } from './middleware/TransactionNotifier'
+import { TransactionWatcher } from './middleware/TransactionWatcher'
+
+use(new Squash())
```

Comment:
Search for `startEffects` in our code base. Following that pattern will make this file support HMR.